gpbft/api.go

```diff
@@ -92,4 +92,6 @@ type Host interface {
 	Signer
 	Verifier
 	DecisionReceiver
+
+	MarshalPayloadForSigning(*Payload) []byte
```

Comment:
I added this as a new function on the base `Host` because it doesn't really fit on either the `Signer` or the `Verifier`. I can also create a new `SignatureOps` (or something) interface that combines `Signer`, `Verifier`, and `MarshalPayloadForSigning`.

Reply:
I merged it into a `Signatures` interface. Open to suggestions on better names.

Reply:
Bikeshedding on name; alternative suggestions: 
* `Signing`, 
* `SignerVerifier`
* (don't hate me) separate all the sign related stuff into their own package called `signing` and call the interface `Scheme` then in `gpbft` it would read `signing.Scheme

Reply:
I'll leave 3 for a future patch. Between 1, 2, and the current name... I don't have any preference.